cdm/core/src/main/java/ucar/nc2/internal/dataset/CoordinatesHelper.java

```diff
@@ -59,7 +59,7 @@ private CoordinatesHelper(Builder builder, NetcdfDataset ncd) {
     coordAxes = ImmutableList.copyOf(axes);
 
     coordTransforms =
-        builder.coordTransforms.stream().map(ct -> ct.build(ncd)).filter(Objects::nonNull).collect(Collectors.toList());
+        builder.coordTransforms.stream().map(ct -> ct.build(ncd, coordAxes)).filter(Objects::nonNull).collect(Collectors.toList());
```

Comment:
This is probably the most important line, since we're passing in the already calculated coordinate axes, before the NetcdfDataset has them initialized. Since we need the coordinate axes to build the coordinate transforms correctly, we need to pass them in separately.